x-pack/filebeat/tests/integration/status_reporter_test.go

```diff
@@ -0,0 +1,217 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent-client/v7/pkg/client/mock"
+	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
+
+	"github.com/elastic/beats/v7/libbeat/common/reload"
+	lbmanagement "github.com/elastic/beats/v7/libbeat/management"
+	"github.com/elastic/beats/v7/x-pack/filebeat/cmd"
+	"github.com/elastic/beats/v7/x-pack/libbeat/management"
+	"github.com/elastic/beats/v7/x-pack/libbeat/management/tests"
+
+	conf "github.com/elastic/elastic-agent-libs/config"
+)
+
+func TestLogStatusReporter(t *testing.T) {
+	unitOneID := mock.NewID()
+	unitOutID := mock.NewID()
+	token := mock.NewID()
+
+	tests.InitBeatsForTest(t, cmd.Filebeat())
+	tmpDir := t.TempDir()
+	filename := fmt.Sprintf("test-%d", time.Now().Unix())
+	outPath := filepath.Join(tmpDir, filename)
+	t.Logf("writing output to file %s", outPath)
+	err := os.Mkdir(outPath, 0775)
+	require.NoError(t, err)
+	defer func() {
+		err := os.RemoveAll(outPath)
+		require.NoError(t, err)
+	}()
```

Comment:
`t.TempDir()` gets removed at the end of the test, you don't need to do it.
```suggestion
```

From the [docs](https://pkg.go.dev/testing#T.TempDir):
> TempDir returns a temporary directory for the test to use. The directory is automatically removed when the test and all its subtests complete.